django/contrib/admin/options.py

```diff
@@ -534,11 +534,11 @@ def get_inline_instances(self, request, obj=None):
         for inline_class in self.inlines:
             inline = inline_class(self.model, self.admin_site)
             if request:
-                if not (inline.has_add_permission(request) or
+                if not (inline.has_add_permission(request, obj) or
                         inline.has_change_permission(request, obj) or
                         inline.has_delete_permission(request, obj)):
                     continue
-                if not inline.has_add_permission(request):
+                if not inline.has_add_permission(request, obj):
```

Comment:
As mentioned in the ticket we'll want to use an approach similar to a7c256cb5491bf2a77abdff01638239db5bfd9d5 to support `inline.has_add_permission(request)` for a certain time.